### PR TITLE
[Feature]: Automaticaly set imported element userModification

### PR DESCRIPTION
**Pull request by root913**
This PR solves problem of identifying who is the author of object change for manualy triggered imports.

As of now every object that was changed by import had userModification set to 0 (that is system user). 
This change introduce new column "**userOwner**" to queue table that by default is set to 0.
If user manualy trigger import it's user id is passed to queue item and later in import process it is set to element userModification property.
This will not affect automatic/cron imports.

### CODE REVIEW

**Comment by root913** [comment]
I have read the CLA Document and I hereby sign the CLA

File: src/DataSource/Interpreter/AbstractInterpreter.php [hunk 1]
```diff
@@ -23,6 +23,7 @@
 use Pimcore\Log\ApplicationLogger;
 use Pimcore\Log\FileObject;
 use Pimcore\Model\Tool\TmpStore;
+use Pimcore\Tool;
```
Comment (mattamon) [inline src/DataSource/Interpreter/AbstractInterpreter.php:26]:
```suggestion
use Pimcore\Tool\Admin;
```

File: src/DataSource/Interpreter/AbstractInterpreter.php [hunk 2]
```diff
@@ -212,10 +213,12 @@ protected function processImportRow(array $data)
             $createQueueItem = $this->deltaChecker->hasChanged($this->configName, $this->idDataIndex, $data);
         }
 
+        $userOwner = Tool\Admin::getCurrentUser()?->getId() ?? 0;
```
Comment (mattamon) [inline src/DataSource/Interpreter/AbstractInterpreter.php]:
```suggestion
        $userOwner = Admin::getCurrentUser()?->getId() ?? 0;
```

Comment (mattamon) [reply src/DataSource/Interpreter/AbstractInterpreter.php]: Could you please add a comment saying that if there is no user logged in we use the system user as `userOwner`

**Review by mattamon — CHANGES_REQUESTED** [review]
Hey @root913,

your PR looks good to me! Just some minor changes. I would always rather import the whole class instead of importing just a part like with Tool\Admin.

Also I would add a comment saying that the 0 is system user just to have it there. 

But other than that it looks solid!

Thank you for your contribution!

**Comment by root913** [comment]
I noticed that latest migration is specified in Installer class (method: getLastMigrationVersionClassName) - should i update it also?

**Comment by mattamon** [comment]
@root913 

> I noticed that latest migration is specified in Installer class (method: getLastMigrationVersionClassName) - should i update it also?

That would be awesome yeah!

File: src/Queue/QueueService.php [hunk 3]
```diff
@@ -86,6 +87,7 @@ protected function createQueueTableIfNotExisting(\Closure $callable = null)
         $this->getDb()->executeQuery(sprintf('CREATE TABLE IF NOT EXISTS %s (
             id bigint AUTO_INCREMENT,
             timestamp bigint NULL,
+            userOwner int unsigned NOT NULL DEFAULT \'0\',
```
Comment (mattamon) [inline src/Queue/QueueService.php]:
```suggestion
            userOwner int unsigned NOT NULL DEFAULT 0,

```

I think we do not need the escaping here for 0 right?

**Review by mattamon — APPROVED** [review]
@root913 Thank you for all the changes and your contribution! This looks good to me!

**Pull request merged**